packages/runtime/src/templates/server.ts

```diff
@@ -52,6 +56,30 @@ const getNetlifyNextServer = (NextServer: NextServerType) => {
       }
     }
 
+    // doing what they do in https://github.com/vercel/vercel/blob/1663db7ca34d3dd99b57994f801fb30b72fbd2f3/packages/next/src/server-build.ts#L576-L580
+    private netlifyPrebundleReact(path: string) {
+      const routesManifest = this.getRoutesManifest()
+      const appPathsManifest = this.getAppPathsManifest()
+
+      const routes = [...routesManifest.staticRoutes, ...routesManifest.dynamicRoutes]
+      const matchedRoute = routes.find((route) => new RegExp(route.regex).test(path))
+      const isAppRoute =
+        appPathsManifest && matchedRoute ? appPathsManifest[joinPaths(matchedRoute.page, 'page')] : false
+
+      if (isAppRoute) {
+        // app routes should use prebundled React
+        // eslint-disable-next-line no-underscore-dangle
+        process.env.__NEXT_PRIVATE_PREBUNDLED_REACT = this.nextConfig.experimental?.serverActions
+          ? 'experimental'
+          : 'next'
+        return
+      }
+
+      // pages routes should use use node_modules React
+      // eslint-disable-next-line no-underscore-dangle
+      process.env.__NEXT_PRIVATE_PREBUNDLED_REACT = ''
```

Comment:
Does this need to be set at all if it's an empty string? OR does Next.js check for an empty string vs. `undefined`?

Reply:
strangely, it doesn't work if it's `undefined` - i'm not sure why and i gave up splelunking to find out!